cpp/Makefile

```diff
@@ -10,21 +10,34 @@ ifeq ($(USE_ASAN), True)
 	use_asan = ${USE_ASAN}
 endif
 
+# JDK path for JNI compilation
+JAVA_HOME ?= /usr/lib/jvm/default-java
+ifneq ($(JDK_PATH),)
+	JAVA_HOME = ${JDK_PATH}
+endif
+
 build:
 	mkdir -p build && cd build && \
-	conan install .. --build=missing -s build_type=Release --update \
-	-o with_ut=True -o with_asan=${use_asan} && \
-	conan build .. && \
-	conan install .. --install-folder . -g make
+	JAVA_HOME=${JAVA_HOME} conan install .. --build=missing -s build_type=Release --update \
+	-o with_ut=True -o with_asan=${use_asan} -o with_jni=True && \
+	JAVA_HOME=${JAVA_HOME} conan build .. && \
+	JAVA_HOME=${JAVA_HOME} conan install .. --install-folder . -g make
 
 package: build
 	mkdir -p build && cd build && \
 	conan export .. milvus-storage/0.1.0@milvus/dev
 
 debug:
 	mkdir -p build && cd build && \
-	conan install .. --build=missing -s build_type=Debug -o with_asan=True && \
-	conan build ..
+	JAVA_HOME=${JAVA_HOME} conan install .. --build=missing -s build_type=Debug -o with_asan=True -o with_jni=True && \
+	JAVA_HOME=${JAVA_HOME} conan build ..
+
+build-no-jni:
```

Comment:
we can control the option is `conanfile`, i am not sure why we need this target?